providers: support for Proxmox VE (continuation)

arcln opened the pull request:
Hello, this PR is a continuation of #1790. The original PR is doing good, but we cannot push to it and we would like things to go faster so we decided to make another one to finish it.

This PR includes the commit from @b-, and a fix for https://github.com/coreos/ignition/pull/1790#discussion_r1674311256. I also added release note.

I think this is good to go. Feel free to tell me otherwise.

jlebon commented on docs/release-notes.md:
```diff
@@ -11,6 +11,7 @@ nav_order: 9
 ### Features
 
 - Support partitioning disk with mounted partitions
+- Added Proxmox VE provider
```
```suggestion
- Support Proxmox VE
```

jlebon replied:
Let's also update `docs/supported-platforms.md` (rendered at https://coreos.github.io/ignition/supported-platforms/).

jlebon reviewed (approved):
Thanks! Some minor docs-related comments, but LGTM otherwise.

Were you able to test this?

jlebon commented:
Also, minor/optional: let's squash it all to one commit (and add yourself as co-author) ?

arcln commented:
I tested this PR and it seems to work fine in various situations. 

I patched the release notes and added Promox VE to supported platforms. I also squashed the PR to a single commit.

abuisine commented:
@jlebon would you be so kind as to have a look ?

jlebon reviewed (approved):
LGTM, thanks!

b- commented:
Awesome! 🚀

abuisine commented:
Exciting ❤️

genebean commented:
Can any of you elaborate on how to put this to use? I am new to FCOS & ignition and may simply not know where I should be looking. Thanks!

jlebon commented:
@genebean If you're interested in using this in FCOS, the larger effort is tracked in https://github.com/coreos/fedora-coreos-tracker/issues/736 and https://github.com/coreos/fedora-coreos-tracker/issues/1652.

lperdereau commented:
Hi, do we know when a version will include this cool feature?
